runtime/src/bank.rs

```diff
@@ -2477,15 +2483,29 @@ impl Bank {
             .feature_set
             .is_active(&feature_set::update_rewards_from_cached_accounts::id());
 
-        self.pay_validator_rewards_with_thread_pool(
-            prev_epoch,
-            validator_rewards,
-            reward_calc_tracer,
-            self.credits_auto_rewind(),
-            thread_pool,
-            metrics,
-            update_rewards_from_cached_accounts,
-        );
+        if self
+            .feature_set
+            .is_active(&feature_set::update_rewards_no_join::id())
+        {
```

Comment:
Does this code change the calculations result that is feature gated? If so, then how does it change?

If this is not supposed to change the result, then it shouldn't be feature gated. Putting it behind the feature will prevent erroneous results to be discovered when running master code on mainnet.

Reply:
After looking at the code, it seems that `rewards` member was not included in the network consensus. 

The only place that depends on `rewards` member is the `RewardsRecoderService`, which saves the `rewards` in blockstore, And I think RPC method relies on the info for reward query. 

@CriesofCarrots Do we have the freedom to change the order of `rewards` without a feature gate?

Reply:
We don't guarantee order in RPC rewards list, so no concerns from that perspective.
A change that adjusts order but not specific elements should be fine to do without a feature gate... famous last words, I'm sure 😉 . Fortunately, when a canary runs this over an epoch boundary, I imagine we'll know immediately if that conclusion is wrong.